documentation/docs/02-layouts.md

```diff
@@ -62,11 +62,115 @@ We can create a layout that only applies to pages below `/settings` (while inher
 <slot></slot>
 ```
 
-### Resets
+### Named layouts
 
-To reset the layout stack, create a `__layout.reset.svelte` file instead of a `__layout.svelte` file. For example, if you want your `/admin/*` pages to _not_ inherit the root layout, create a file called `src/routes/admin/__layout.reset.svelte`.
+Some parts of your app might need something other than the default layout. For these cases you can create _named layouts_...
 
-Layout resets are otherwise identical to normal layout components.
+```svelte
+/// file: src/routes/__layout-foo.svelte
+<div class="foo">
+	<slot></slot>
+</div>
+```
+
+...and then use them by referencing the layout name (`foo`, in the example above) in the filename:
+
+```svelte
+/// file: src/routes/my-special-page@foo.svelte
+<h1>I am inside __layout-foo</h1>
+```
+
+Named layouts are very powerful, but it can take a minute to get your head round them. Don't worry if this doesn't make sense all at once.
+
+#### Scoping
+
+Named layouts can be created at any depth, and will apply to any components in the same subtree. For example, `__layout-foo` will apply to `one` and `two`, but not `three` or `four`:
+
+```
+src/routes/
+├ a/
+│ ├ b/
+│ │ ├ c/
+│ │ │ ├ __layout-foo.svelte
+│ │ │ ├ one@foo.svelte
```

Comment:
what if someone has a route like `folding@home.svelte` where they don't want the `@` to represent a named layout? should we add a mention here of some way to escape the `@` symbol or use route matchers?

Reply:
i've opened https://github.com/sveltejs/kit/issues/4403